keck_vnc_launcher.py

```diff
@@ -618,13 +632,13 @@ def launch_vncviewer(self, vncserver, port, geometry=None):
         vncargs = self.config.get('vncargs', None)
 
         cmd = [vncviewercmd]
-        if vncargs:
+        if vncargs is not None:
             vncargs = vncargs.split()
             cmd = cmd + vncargs
-        if self.args.viewonly:
+        if self.args.viewonly is not None:
             cmd.append('-ViewOnly')
         #todo: make this config on/off so it doesn't break things
-        if geometry:
+        if geometry is not None:
```

Comment:
This causes crash because geometry='' coming in.

Reply:
I modified the check to include looking for the empty string, but it seems like the caller should provide None instead of '' if they aren't specifying the geometry. That would be more consistent with how the function is defined. Still, it doesn't hurt to make the condition more explicit.

Reply:
Should now be fixed.

I'm fine with the idea of testing a few different use cases; what's in this branch all works for me, but I can't prove to you which conditions I'm exercising.

Reply:
I'll help look at them all with a closer eye sometime tomorrow.  We don't have a test suite to test all of these conditionals so there is a small risk there could be more like this.